circuits/src/mpc_circuits/settle.rs

```diff
@@ -44,20 +53,61 @@ where
     let party0_buy = amounts.remove(0);
     let party0_sell = amounts.remove(0);
 
+    // Compute the fees paid by each party
+    let (party0_fees, party1_fees) =
+        compute_settlement_fees(relayer_fee0, relayer_fee1, &party0_buy, &party0_sell, fabric);
+
     // Update the balances of the two parties
+    let party0_net = &party0_buy - party0_fees.total();
     let party0_buy_balance = &mut party0_new_shares.balances[party0_settle_indices.balance_receive];
-    party0_buy_balance.amount = &party0_buy_balance.amount + &party0_buy;
+    party0_buy_balance.amount = &party0_buy_balance.amount + &party0_net;
+    party0_buy_balance.relayer_fee_balance =
+        &party0_buy_balance.relayer_fee_balance + &party0_fees.relayer_fee;
+    party0_buy_balance.protocol_fee_balance =
+        &party0_buy_balance.protocol_fee_balance + &party0_fees.protocol_fee;
 
+    let party1_net = &party0_sell - party1_fees.total();
     let party1_buy_balance = &mut party1_new_shares.balances[party1_settle_indices.balance_receive];
-    party1_buy_balance.amount = &party1_buy_balance.amount + &party0_sell;
+    party1_buy_balance.amount = &party1_buy_balance.amount + &party1_net;
+    party1_buy_balance.relayer_fee_balance =
+        &party1_buy_balance.relayer_fee_balance + &party1_fees.relayer_fee;
+    party1_buy_balance.protocol_fee_balance =
+        &party1_buy_balance.protocol_fee_balance + &party1_fees.protocol_fee;
 
     let party0_sell_balance = &mut party0_new_shares.balances[party0_settle_indices.balance_send];
     party0_sell_balance.amount = &party0_sell_balance.amount - &party0_sell;
 
     let party1_sell_balance = &mut party1_new_shares.balances[party1_settle_indices.balance_send];
     party1_sell_balance.amount = &party1_sell_balance.amount - &party0_buy;
 
-    (party0_new_shares, party1_new_shares)
+    (party0_fees, party1_fees, party0_new_shares, party1_new_shares)
+}
+
+/// Compute the fee take for each party in a settlement
+fn compute_settlement_fees(
+    relayer_fee0: &AuthenticatedFixedPoint,
+    relayer_fee1: &AuthenticatedFixedPoint,
+    party0_buy_amount: &AuthenticatedScalar,
+    party1_buy_amount: &AuthenticatedScalar,
+    fabric: &Fabric,
+) -> (AuthenticatedFeeTake, AuthenticatedFeeTake) {
+    // Relayer fees
+    let party0_relayer_fee_fp = relayer_fee0 * party0_buy_amount;
+    let party0_relayer_fee = FixedPointMpcGadget::as_integer(&party0_relayer_fee_fp, fabric);
+    let party1_relayer_fee_fp = relayer_fee1 * party1_buy_amount;
+    let party1_relayer_fee = FixedPointMpcGadget::as_integer(&party1_relayer_fee_fp, fabric);
+
+    // Protocol fees
+    let protocol_fee = *PROTOCOL_FEE_FP;
```

Comment:
This is ok for now, but just calling out that down the line we should read the protocol fee from the contract